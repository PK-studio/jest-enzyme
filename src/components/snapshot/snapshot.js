import React from 'react'

export const Snapshot = ({isFeatureToggled}) => (
    <>
        <h1>Test name: snapshot</h1>
        {isFeatureToggled
            ? <p>Feature is switch on</p>
            : <p>This time feature is switch off</p>
        }
    </>
)