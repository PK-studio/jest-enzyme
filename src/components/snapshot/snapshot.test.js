import React from 'react'
import { Snapshot } from './snapshot'

describe('Snapshot', () => {
    it('Renders component with switch off feature', () => {
        const component = mount(<Snapshot />)
        expect(component).toMatchSnapshot()
    })

    it('Renders info when feature is on', () => {
        const component = shallow(<Snapshot isFeatureToggled/>)
        const searchingText = 'Feature is switch on'
        expect(component.text().includes(searchingText)).toBe(true)
    })

    it('Return link on click', () => {
        const mockOnClick = jest.fn()
        const mockLink = 'http://test/link'
        const component = shallow(<Snapshot link={mockLink} onClick={mockOnClick} />)
        const button = component.find('[data-test-id="snapshot-button"]')
        button.simulate('click')
        expect(mockOnClick).toBeCalledTimes(1)
        expect(mockOnClick).toBeCalledWith(mockLink)
    })
})