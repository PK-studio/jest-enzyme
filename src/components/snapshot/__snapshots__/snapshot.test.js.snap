// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Renders component with switch off feature 1`] = `
<Snapshot
  isFeatureToggled={false}
>
  <h1>
    Test name: snapshot
  </h1>
  <p>
    This time feature is switch off
  </p>
</Snapshot>
`;
